### ISSUE TITLE
Protocol Engine: Support pick up tip command mapping

### ISSUE THREAD

**Issue by shlokamin** [Opentrons/opentrons#8717]
# Overview

For generating labware position check commands, we depend on inspecting a `pickUpTip` command's pipette id and labware id to know which labware a pipette interacts with. 

This means that in a resolved protocol analysis we need pick up tip commands to be resolved to a [`PickUpTipCommand` interface](https://github.com/Opentrons/opentrons/blob/d790a0bd04e35b78d93526850d90cf4d46db91cc/shared-data/protocol/types/schemaV6/command/pipetting.ts#L42): 

```js
{
      "id": "some_id",
      "commandType": "pickUpTip",
      "params": {
        "pipetteId": "pipetteId",
        "labwareId": "pipetteId",
        "wellName": "A1"
      }
}
```

# Acceptance criteria

Resolved protocol analysis pick up tip commands match the interface above

**Comment by SyntaxColoring** [comment]
This might be tricky because the `pickUpTip` command needs a correct pipette ID and tip rack ID. The legacy contexts can't easily report those to our legacy mapper because they don't know about Protocol Engine IDs.

We could maybe bring Protocol Engine IDs into the legacy contexts. Or, alternatively:

* Legacy mapper will keep track of which of the labware ID is in each deck slot
* Legacy mapper will keep track of which pipette ID is in each mount
* Legacy contexts should know, and should easily be able to report, the deck slot and pipette mount of a tip pickup
* Legacy mapper matches those against its memory of IDs

In other words, to avoid bringing Protocol Engine's ideas of labware and pipette IDs into the legacy contexts, we can rely on the fact that there are only ever two pipette mounts, and there is at most one labware in each deck slot.

(Thanks @b-cooper and @shlokamin for brainstorming.)
